src/FacebookAds/Object/AbstractObject.php

```diff
@@ -45,6 +45,8 @@ public function __construct() {
   public function __get($name) {
     if (array_key_exists($name, $this->data)) {
       return $this->data[$name];
+    } elseif (in_array($name, static::getFieldsEnum()->getValues())) {
```

Comment:
The logic looks good, but the enum already offer an index based validation function that will perform better than in_array.

can you replace

~~~php
in_array($name, static::getFieldsEnum()->getValues())
~~~

with

~~~php
static::getFieldsEnum()->isValid($name)
~~~

?